cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -108,8 +108,8 @@ func (ng *nodegroup) DeleteNodes(nodes []*corev1.Node) error {
 	}
 
 	// Step 2: if deleting len(nodes) would make the replica count
-	// <= 0, then the request to delete that many nodes is bogus
-	// and we fail fast.
+	// < node group minimum size, then the request to delete that many
```

Comment:
this updates the comment but I can't see any change in code?

Reply:
good spot, i need to remove that. i forgot to when i changed the code.

Reply:
you saying we don't want to consolidate the code to reflect what the code does any more but remove the undo the comment? It was making sense to me change it to compare against `MinSize`

Reply:
i think the MinSize change is good, it is an improvement over what we had. but @michaelgugino got into a discussion about it, and he preferred that i revert the MinSize change since it didn't affect the root cause here. furthermore, he wanted to see more proof that the MinSize change was needed.

i am now trying to prove whether we need this check by looking at the autoscale deletion mechanisms. i'm not sure the best way to proceed here, but since the core of this issue is related to the Unmark mechanism i figured we could wait to make the MinSize change since there was an objection.

Reply:
ok, It seems a reasonable change to me. If we don't want to include here we should put a another PR for it to discuss. At minimum we should consider to compare minSize against current and short circuit there, see:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go#L273-L275

https://github.com/kubernetes/autoscaler/blob/babcd6121a436a05a3f03ce810ae08dac0a880da/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go#L458-L459

https://github.com/kubernetes/autoscaler/blob/0690933b9d0b9ef956dc6ec1928070ad21ac5c81/cluster-autoscaler/cloudprovider/packet/packet_node_group.go#L132

 https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go#L273-L276

Reply:
++, totally agree

i will create another pull request with just the minsize change. in addition to those links you provided, this from the core is also good supporting evidence:

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/scale_down.go#L842